tools/build_executable.py

```diff
@@ -8,85 +8,107 @@
         target_platform = "Mac"
-        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}-Unsigned")
+        archive_dir     = os.path.realpath(os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}-Unsigned"))
+        unreal_tmp_dir  = os.path.expanduser(os.path.join("~", "Library", "Preferences", "Unreal Engine", "SpearSimEditor"))
+
         if args.conda_script:
             conda_script = args.conda_script
         else:
             # default for graphical install, see https://docs.anaconda.com/anaconda/user-guide/faq/
-            conda_script = os.path.join("~", "opt", "anaconda3", "etc", "profile.d", "conda.sh")
+            conda_script = os.path.expanduser(os.path.join("~", "opt", "anaconda3", "etc", "profile.d", "conda.sh"))
+
         cmd_prefix = f". {conda_script}; conda activate {args.conda_env}; "
 
     elif sys.platform == "linux":
-        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        run_uat_script  = os.path.realpath(os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh"))
         target_platform = "Linux"
-        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+        archive_dir     = os.path.realpath(os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}"))
+        unreal_tmp_dir  = None
+
         if args.conda_script:
             conda_script = args.conda_script
         else:
             # see https://docs.anaconda.com/anaconda/user-guide/faq/
-            conda_script = os.path.join("~", "anaconda3", "etc", "profile.d", "conda.sh")
+            conda_script = os.path.expanduser(os.path.join("~", "anaconda3", "etc", "profile.d", "conda.sh"))
+
         cmd_prefix = f". {conda_script}; conda activate {args.conda_env}; "
 
     else:
         assert False
 
-    # clone repo along with submodules
+    # remove our temporary repository directory to ensure a clean build
+    if os.path.exists(repo_dir):
+        print(f"[SPEAR | build_executable.py] Repository exists, removing: {repo_dir}")
+        shutil.rmtree(repo_dir)
+        os.makedirs(repo_dir)
+
+    # We need to remove this temp dir (created by the Unreal build process) because it contains paths from previous builds.
+    # If we don't do this step, we will get many warnings during this build:
+    #     Warning: Unable to generate long package name for path/to/previous/build/Some.uasset because FilenameToLongPackageName failed to convert
+    #     'path/to/previous/build/Some.uasset'. Attempt result was '../../../../../../path/to/previous/build/path/to/previous/build/Some', but the
+    #     path contains illegal characters '.'
+    print(unreal_tmp_dir)
+    print()
+    if os.path.exists(unreal_tmp_dir):
+        print(f"[SPEAR | build_executable.py] Global cache directory exists, removing: {unreal_tmp_dir}")
```

Comment:
Should we add `unreal engine` to the print statement? This way it makes it more clear that the global cache directory we reference here is managed by Unreal. Something like
```python
print(f"[SPEAR | build_executable.py] Unreal Engine's global cache directory exists, removing: {unreal_tmp_dir}")
```

Reply:
Agreed.